ssc/cmod_host_developer.cpp

```diff
@@ -2657,16 +2707,55 @@ class cm_host_developer : public compute_module
+        escal_or_annual(CF_util_escal_rate, nyears, "rate_escalation", inflation_rate, 0.01, true, 0);
+    save_cf(CF_util_escal_rate, nyears, "cf_util_escal_rate");
+    cf.at(CF_parasitic_cost, 0) = 0;
+    double monthly_hourly_purchases = 0;
+    double monthly_hourly_fromgrid = 0;
+    for (int a = 1; a <= nyears; a++) {
+        for (size_t m = 1; m <= 12; m++) {
+            monthly_hourly_purchases = 0;
+            monthly_hourly_fromgrid = 0;
+            for (size_t d = 1; d <= util::days_in_month(int(m - 1)); d++) {
+                for (size_t h = 0; h < 24; h++) { //monthly iteration for each year
+                    if (year1_hourly_e_from_grid[h] != 0.0) {
+                        //cf.at(CF_charging_cost_grid_month, a) += monthly_grid_to_batt[m] / (monthly_grid_to_batt[m] + monthly_grid_to_load[m]) * monthly_energy_charge[m] * charged_grid[a] / charged_grid[1] * cf.at(CF_util_escal_rate, a);
+                        monthly_hourly_purchases += -hourly_energy_calcs.hourly_purchases()[util::hour_of_year(m, d, h)] * cf.at(CF_degradation, a);
+                        monthly_hourly_fromgrid += year1_hourly_e_from_grid[h];
+                        cf.at(CF_parasitic_cost, a) += -hourly_energy_calcs.hourly_purchases()[util::hour_of_year(m, d, h)] * cf.at(CF_degradation, a) / year1_hourly_e_from_grid[h] * (year1_hourly_ec[h] + year1_hourly_dc[h]) * cf.at(CF_util_escal_rate, a); //use the electricity rate data by year (also trueup) //* charged_grid[a] / charged_grid[1] * cf.at(CF_util_escal_rate, a);
+                    }
+                    if (d == util::days_in_month(int(m - 1)) && h == 23) cf.at(CF_parasitic_cost, a) += -monthly_gen_purchases[(size_t(a) - 1) * 12 + m - 1] / monthly_e_fromgrid[m - 1] * (net_annual_true_up.at(a, m) + net_billing_credit.at(a, m) + net_metering_credit.at(a, m));
+
+                }
+            }
+
+        }
+    }
 
+    save_cf(CF_parasitic_cost, nyears, "cf_parasitic_cost");
 
-	double host_npv_energy_real = npv(CF_energy_net, nyears, host_disc_real);
-	double host_lcoe_real = -(cf.at(CF_after_tax_net_equity_cost_flow, 0) + npv(CF_after_tax_net_equity_cost_flow, nyears, host_nom_discount_rate)) * 100;
+	double host_npv_energy_real = npv(CF_energy_sales, nyears, host_disc_real);
+	double host_lcoe_real = -(cf.at(CF_after_tax_net_equity_cost_flow, 0) + npv(CF_after_tax_net_equity_cost_flow, nyears, host_nom_discount_rate) + npv(CF_parasitic_cost, nyears, nom_discount_rate)) * 100;
 	if (host_npv_energy_real == 0.0)
 		host_lcoe_real = std::numeric_limits<double>::quiet_NaN();
 	else
 		host_lcoe_real /= host_npv_energy_real;
 
-	double host_npv_energy_nom = npv(CF_energy_net, nyears, host_nom_discount_rate);
+	double host_npv_energy_nom = npv(CF_energy_sales, nyears, host_nom_discount_rate);
 	double host_lcoe_nom = -(cf.at(CF_after_tax_net_equity_cost_flow, 0) + npv(CF_after_tax_net_equity_cost_flow, nyears, host_nom_discount_rate)) * 100;
```

Comment:
I see where the parasitics are included in the real LCOE, but what about nominal?